docs/src/content/en/docs/agents/dynamic-agents.mdx

```diff
@@ -0,0 +1,47 @@
+---
+
+title: Dynamic Agents | Agents | Mastra Docs
+description: Dynamically configure your agent's instruction, model and tools using runtime context.
+
+# Dynamic Agents
+
+Dynamic agents use [runtime context](./runtime-variables), like user IDs and other important parameters, to adjust their settings in real-time. This means they can change the model they use, update their instructions, and select different tools as needed. By using this context, agents can better respond to each user's needs. They can also call any API to gather more information, which helps improve what the agents can do.
```

Comment:
Some ideas for a follow-up:

-----
It's often the case that real-world agents need to be customized on a per-user basis: they may need access to user-specific data, API keys, or other runtime context.

Let's say you want to let your user choose the model, or you want to customize the system prompt on a per-user basis, or customize an agent's toolsets on a per-request basis.

Mastra lets you do this with dynamic agents, which use runtime context....